bentoml/_internal/frameworks/catboost.py

```diff
@@ -0,0 +1,273 @@
+            ``{"my-normalizer": normalizer}``.
+
+            Custom objects are currently serialized with cloudpickle, but this implementation is
+            subject to change.
+        metadata (``dict[str, Any]``, optional):
+            Metadata to be associated with the model. An example is ``{"max_depth": 2}``.
+
+            Metadata is intended for display in model management UI and therefore must be a default
+            Python type, such as ``str`` or ``int``.
+    Returns:
+        :obj:`~bentoml.Tag`: A :obj:`tag` with a format `name:version` where `name` is the
+        user-defined model's name, and a generated `version` by BentoML.
+
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+        import numpy as np
+
+        from catboost import CatBoostClassifier, Pool
+
+        # initialize data
+        train_data = np.random.randint(0, 100, size=(100, 10))
+
+        train_labels = np.random.randint(0, 2, size=(100))
+
+        test_data = catboost_pool = Pool(train_data, train_labels)
+
+        model = CatBoostClassifier(iterations=2,
+                                   depth=2,
+                                   learning_rate=1,
+                                   loss_function='Logloss',
+                                   verbose=True)
+        # train the model
+        model.fit(train_data, train_labels)
+
+        # save the model to the BentoML model store
+        bento_model = bentoml.catboost.save_model("my_catboost_model", model)
+    """  # noqa: LN001
+    if not isinstance(model, cb.CatBoost):
+        raise TypeError(f"Given model ({model}) is not a catboost.CatBoost.")
+
+    context: ModelContext = ModelContext(
+        framework_name="catboost",
+        framework_versions={"catboost": get_pkg_version("catboost")},
+    )
+
+    if signatures is None:
+        logger.info(
+            'Using default model signature `{"predict": {"batchable": False}}` for CatBoost model'
+        )
+        signatures = {
+            "predict": {"batchable": False},
+        }
+
+    if metadata is None:
+        metadata = {}
+
+    metadata[MODEL_TRAINING_CLASS_NAME_KEY] = model.__class__.__name__
+
+    with bentoml.models.create(
+        name,
+        module=MODULE_NAME,
+        api_version=API_VERSION,
+        signatures=signatures,
+        labels=labels,
+        custom_objects=custom_objects,
+        metadata=metadata,
+        context=context,
+    ) as bento_model:
+        model.save_model(bento_model.path_of(MODEL_FILENAME))  # type: ignore (incomplete CatBoost types)
+
+        return bento_model
+
+
+def get_runnable(bento_model: bentoml.Model) -> t.Type[bentoml.Runnable]:
+    """
+    Private API: use :obj:`~bentoml.Model.to_runnable` instead.
+    """
+
+    class CatBoostRunnable(bentoml.Runnable):
+        SUPPORT_NVIDIA_GPU = True
+        SUPPORT_CPU_MULTI_THREADING = True
+
+        predict_params: t.Dict[str, t.Any]
```

Comment:
should we refactor this to ModelOptions? @ssheng

Reply:
@aarnphm There is no need for these values to be persisted in the model, as they are only useful in the runner and depend on runtime environment variables